### PR TITLE
Marshal the JSON correctly in verify, and properly load keys.

### PR DESCRIPTION
**Pull request by dlorenc**

### CODE REVIEW

**Comment by lukehinds** [comment]
LGTM!

File: pkg/types/types.go [hunk 1]
```diff
@@ -152,6 +158,10 @@ func (r *RekorEntry) Load(ctx context.Context) error {
 		dataReader = bytes.NewReader(r.Data)
 	}
 
+	if err := r.ParseKeys(); err != nil {
```
Comment (bobcallaway) [inline pkg/types/types.go:161]: I'd move this call before the URL fetch. There's no reason to even call `Get()` if we don't have a valid key/sig pair.

File: pkg/types/types.go [hunk 2]
```diff
@@ -90,16 +97,15 @@ func ParseRekorLeaf(r io.Reader) (*RekorLeaf, error) {
...
 	if err != nil {
-		return nil, err
+		return err
 	}
-
-	return &l, nil
+	return nil
 }
 
 func ParseRekorEntry(r io.Reader, leaf *RekorLeaf) (*RekorEntry, error) {
```
Comment (bobcallaway) [inline pkg/types/types.go:111]: We should validate the pointer passed in here as well.

File: pkg/types/types.go [hunk 3]
```diff
@@ -77,10 +77,17 @@ func ParseRekorLeaf(r io.Reader) (*RekorLeaf, error) {
 		return nil, err
 	}
 
+	if err := l.ParseKeys(); err != nil {
+		return nil, err
+	}
+	return &l, nil
+}
+
+func (l *RekorLeaf) ParseKeys() error {
```
Comment (bobcallaway) [inline pkg/types/types.go:86]: Since you're also "validating" the SHA, perhaps name it something like "ValidateLeaf"?

**Pull request merged**